parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1151,6 +1152,11 @@ public ReadBuilder withAADPrefix(ByteBuffer aadPrefix) {
       return this;
     }
 
+    public ReadBuilder pushedlimit(int limit) {
```

Comment:
Should we assert this is greater than 1? I assume and input 0 or negative is a bad call

Reply:
I think we are OK because Spark has assert for limit. If the limit is negative, e.g. `SELECT * FROM table limit -2`, Spark will throw 
```
org.apache.spark.sql.AnalysisException: [INVALID_LIMIT_LIKE_EXPRESSION.IS_NEGATIVE] The limit like expression "-2" is invalid. The limit expression must be equal to or greater than 0, but got -2.;
```
If the limit is 0, e.g. `SELECT * FROM table limit 0`, Spark changes this to an empty table scan and it won't reach here.

Reply:
This is public though, and anyone can call it so we can't rely on Spark to cover us.

Reply:
Right, I forgot that. I've added a check to throw an `IllegalArgumentException` if the limit is <= 0

Reply:
Usually we do     `Preconditions.checkArgument`
for things like this

Reply:
Changed to `Preconditions.checkArgument`. Thanks